addons/hr_holidays_attendance/report/hr_leave_attendance_report.py

```diff
@@ -0,0 +1,216 @@
+
+
+class HrLeaveAttendanceReport(models.Model):
+    _name = "hr.leave.attendance.report"
+    _description = "Attendance and Leave Analysis Report"
+    _auto = False
+
+    def _compute_display_name(self):
+        for rec in self:
+            rec.display_name = f"{rec.employee_id.display_name}, {rec.date}"
+
+    date = fields.Date("Date")
+    employee_id = fields.Many2one("hr.employee", string="Employee")
+    schedule_id = fields.Many2one("resource.calendar", string="Working Schedule")
+    expected_hours = fields.Float("Expected Hours")
+    worked_hours = fields.Float("Worked Hours")
+    leave_hours = fields.Float("Approved Time Off")
+    difference_hours = fields.Float("Difference", help="Worked Hours - Expected Hours + Approved Time Off")
+    leave_ids = fields.Many2many("hr.leave", string="Time Offs", compute="_compute_leave_fields", readonly=True)
+    leave_type_names = fields.Char(
+        string="Time Off Types",
+        compute="_compute_leave_fields",
+    )
+    attendance_ids = fields.Many2many(
+        "hr.attendance",
+        string="Attedances",
+        compute="_compute_attendance_ids",
+        readonly=True,
+    )
+
+    @api.depends('employee_id', 'date')
+    def _compute_leave_fields(self):
+        MIN_DATE, MAX_DATE = self._get_min_max_day()
+        leaves_by_employees = dict(self.env['hr.leave']._read_group(
+            domain=[
+                ('employee_id', 'in', self.employee_id.ids),
+                ('state', '=', 'validate'),
+                ('date_from', '<=', MAX_DATE),
+                ('date_to', '>=', MIN_DATE),
+            ],
+            groupby=['employee_id'],
+            aggregates=['id:recordset'],
+        ))
+        for rec in self:
+            leaves = leaves_by_employees.get(rec.employee_id, self.env['hr.leave'])
+            rec_date_leaves = leaves.filtered(
+                lambda lv: self._timestamped(lv.date_from) <= rec.date <= self._timestamped(lv.date_to),
+            )
+            rec.leave_ids = rec_date_leaves.mapped('id')
+            leave_type_ids = rec_date_leaves.mapped('holiday_status_id')
+            rec.leave_type_names = ', '.join(leave_type_ids.mapped('name'))
+
+    @api.depends('employee_id', 'date')
+    def _compute_attendance_ids(self):
+        MIN_DATE, MAX_DATE = self._get_min_max_day()
+        attendances_by_employees = dict(self.env['hr.attendance']._read_group(
+            domain=[
+                ('employee_id', 'in', self.employee_id.ids),
+                ('check_in', '>=', MIN_DATE),
+                ('check_in', '<=', MAX_DATE),
+            ],
+            groupby=['employee_id'],
+            aggregates=['id:recordset'],
+        ))
+        for rec in self:
+            attendances = attendances_by_employees.get(rec.employee_id, self.env['hr.attendance'])
+            rec.attendance_ids = attendances.filtered(
+                lambda att: self._timestamped(att.check_in) == rec.date,
+            ).mapped('id') or False
```

Comment:
I think `.ids` there and a empty is same as False  
```suggestion
            ).ids
```